timed/reports/filters.py

```diff
@@ -72,29 +70,25 @@ def filter_cost_center(self, queryset, name, value):
             # Customer mode: We only need to filter aggregates,
             # as the customer has no cost center
             return queryset.filter_aggregate(filter_q)
-        else:
-            # Project or task: Filter both to get the correct result
-            return queryset.filter_base(filter_q).filter_aggregate(filter_q)
+        # Project or task: Filter both to get the correct result
+        return queryset.filter_base(filter_q).filter_aggregate(filter_q)
 
     def filter_queryset(self, queryset):
         qs = super().filter_queryset(queryset)
 
         duration_ref = self._refs["reports_ref"] + "__duration"
 
-        full_qs = qs._base.annotate(
+        full_qs = qs._base.annotate(  # noqa: SLF001
             duration=Coalesce(
-                Sum(duration_ref, filter=qs._agg_filters),
+                Sum(duration_ref, filter=qs._agg_filters),  # noqa: SLF001
```

Comment:
SLF001 is now disabled

Reply:
[SLF001](https://docs.astral.sh/ruff/rules/private-member-access/) feels like a good rule tho, are we sure that _agg_filters really should be private?

Reply:
Yeah generally I think it's alright - but then again, we can get into deep OOP philosophy discussions here:

Are private methods/properties considered private as in the classical OOP, "only accessible within the same class", or would it make sense to expand it as "should only be used by the same feature", where the feature may span across a few places outside of the original class... 🤔

In Python, private just means "please don't touch", not "you cannot, technically, have access to this".

I'm now swinging back and thinking we should enable the rule, but keep the code `#noqa`'d. And think of the rule as such: It's a marker to say "I know what I'm doing" (which was the case when I wrote the code here), but have the linter yell at you if you just happen to mess with private stuff without thinking too much about it

Reply:
i re-enabled it

Reply:
I like using noqa as a hint that we gave this proper thought.

IMO part of a classes contract when using a private variable is also "i wont tell you when i delete this and noticing+fixing this will be your issue". given it's (hopefully) only use is in this codebase this is probably fine